src/components/KBArticleSearch/KBArticleSearch.js

```diff
@@ -1,38 +1,148 @@
 import React from 'react';
 import styles from './styles.module.css'
-import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
-import {useState, useEffect} from "react";
+import {useState, useEffect, useRef} from "react";
+import FlexSearch from 'flexsearch'
+const KBArticleSearch = ({kb_articles, kb_articles_and_tags, onUpdateResults}) => {
+
+    const indexRef = useRef(null);
+    let searchTerm;
+    let setSearchTerm;
+    let searchResults;
+    let setSearchResults;
+    let matchedArticles;
+    let setMatchedArticles;
+
+    if(typeof localStorage !== "undefined") {
```

Comment:
whats the value here? why do we need to persist results in local storage?

Reply:
Has to do with how the docusaurus blog components are setup, there is no single component for the tags page, the recent page and the article page. As a result when you change routes the sidebar re-renders each time. I'm using local storage to persist the current search between page navigations.